src/map.cpp

```diff
@@ -7452,6 +7452,8 @@ void map::draw_fill_background(const id_or_id<ter_t> & f) {
 }
 
 void map::draw_square_ter(ter_id type, int x1, int y1, int x2, int y2) {
```

Comment:
It'd be super nice to replace all these drawing functions with generic or template-based versions of themselves, then you can just do:
```
map::draw_square_ter( ter_id type, point p1, point p2 ) {
    draw_square( p1, p2, [this, type] ( x, y ) {
        ter_set(x, y, type)
    } );
}
```
The line drawing functions wouldn't have anything to do with the map code, so could go in another module, probably a header if they're template based.